### DISCUSSION TITLE
Suppress locale in URL path when using i18n domain routing

### DISCUSSION THREAD

**Discussion by tangochris** [vercel/next.js#38940] [Help]
**Background:** We are using i18n domain routing for a Canada `en-CA` and US `en-US` version of a [website](https://www.documoose.ca). Default is `en-CA`.

**Expected behavior:** Locale never appears in URL path. It is unnecessary, as locale is implied by the domain (`.ca` vs `.com`).

**Actual behavior:** On initial load of `www.documoose.ca` website, it 308 redirects to `www.documoose.ca/en-CA/`. Locale does not appear on any further navigation. Locale does not appear on `en-US` site (www.documoose.com).

next.config.js:
```
module.exports = {
    trailingSlash: true,
    i18n: {
        locales: ['en-CA', 'en-US'],
        defaultLocale: 'en-CA',
        domains: [
            {
                domain: 'www.documoose.com',
                defaultLocale: 'en-US',
            },
            {
                domain: 'www.documoose.ca',
                defaultLocale: 'en-CA',
            },
        ],
        localeDetection: false,
    },
};
```

Are we doing something wrong? Why does the locale appear only for the default locale, and only on initial load of the home page?

**Comment by Babbili** [comment]
@tangochris it shouldn't appear as long it is the default one,
Remove `localeDetection: false` and try
